src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,36 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { setQuery, setStatus } from '../../features/filter';
+import { useAppSelector } from '../../app/hooks';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(setQuery(event.target.value));
+  };
+
+  const handleStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setStatus(event.target.value));
```

Comment:
The `handleStatusChange` function dispatches `setStatus` with `event.target.value`, which is of type `string`. Ensure that `setStatus` can handle a `string` type or convert the value to the appropriate type if necessary. If `Status` is an enum, you might need to cast the value to `Status`.